datastore/mongodb/filter_outputs.go

```diff
@@ -31,47 +31,77 @@ func (c *Client) CreateFilterOutput(ctx context.Context, f *model.FilterOutput)
 func (c *Client) UpdateFilterOutput(ctx context.Context, f *model.FilterOutput) error {
 	col := c.collections.filterOutputs
 
-	var docs []model.FilterOutput
-	sc := bson.M{"id": f.ID}
-
-	var err error
-	var num int
-
-	if num, err = c.conn.Collection(col.name).Find(ctx, sc, &docs, mongodb.Limit(1)); err != nil {
-		return errors.Wrap(err, "failed to update filter output")
-	}
-	if num < 1 {
-		return &er{
-			err:      errors.Errorf("failed to find filter output"),
-			notFound: true,
-		}
+	docs, err := c.findFilterOutput(f.ID, ctx)
+	if err != nil {
+		return err
 	}
-
 	//a record with stat 'completed' can't be updated further
 	if docs[0].State == model.Completed {
-		return &er{
-			err:       errors.Errorf(`filter output is already in "completed" state`),
-			forbidden: true,
+		err := &er{
+			err: errors.Errorf(`filter output is already in "completed" state`),
 		}
+		err.forbidden = true
+		return err
 	}
 
-	updates := bson.M{"$set": bson.M{"state": f.State, "downloads": f.Downloads}}
+	nv := bson.M{"$set": bson.M{"state": f.State, "downloads": f.Downloads}}
 
 	var rec *mongodb.CollectionUpdateResult
+	sc := bson.M{"id": f.ID}
 
-	if rec, err = c.conn.Collection(col.name).Update(ctx, sc, updates); err != nil {
+	if rec, err = c.conn.Collection(col.name).Update(ctx, sc, nv); err != nil {
 		return errors.Wrap(err, "failed to update filter output")
 	}
 
 	//This should not happen unless the recored is being removed btween the first find and the update.
 	//Check if the update failed sue to search condition.
 	//Update returns no error but the result's MatchedCount has to be checked for number of updated records
 	if rec.MatchedCount < 1 {
-		return &er{
-			err:      errors.Errorf("failed to find filter output"),
-			notFound: true,
+		err := &er{
+			err: errors.Errorf("failed to find filter output"),
 		}
+		err.notFound = true
+		return err
+	}
+
+	return nil
+}
+
+//AddFilterOutputEvent will append to the existing list of events in the filter output
+func (c *Client) AddFilterOutputEvent(ctx context.Context, id string, f *model.Event) error {
+	col := c.collections.filterOutputs
+	docs, err := c.findFilterOutput(id, ctx)
```

Comment:
Is it necessary to fetch some filter outputs to reinsert? Is there not away to update with one mongo query as we do elsewhere?